text/0000-export-to-oci.md

```diff
@@ -0,0 +1,213 @@
+  - `library-url` MUST contain the url of the library used to export the image
+  - `library-language` MUST contain the client's library programming language used to export the image
+
+The main idea of this new section in the `report.toml` file is to provide the information of the **expected** digest of the image if it is exported to a registry using a particular implementation. This information can be used by other tools (like publish) to complete the operation and verify the consistency of the image.
+
+# Migration
+[migration]: #migration
+
+<!--
+This section should document breaks to public API and breaks in compatibility due to this RFC's proposed changes. In addition, it should document the proposed steps that one would need to take to work through these changes. Care should be give to include all applicable personas, such as platform developers, buildpack developers, buildpack users and consumers of buildpack images.
+-->
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- We could increase the disk space if we not managed the duplication of saving the layers on disk. Currently the Cache implementation (used when daemon is ON) saved the layers tarballs on disk, because the current proposal is exporting the whole image on disk it will also require more space to save the layers for the OCI format in the `blobs` folder.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Redesign the current Cache
+
+Another potential solution could be to export the OCI image along with the current Cache concept and expose some contract for `Platform` to interact with this Cache and extract the final OCI image. At high level, the solution can be represented with the following container diagram from C4 model
+
+![](https://i.imgur.com/xl4gL1G.png)
+
+Notice that on this solution, because the Cache is an internal component from the Lifecycle implementation we will have to expose some kind of specification for `Platforms` to understand its format and been able to read the OCI image.
+
+The current implementation when the Daemon is enable can be describe with the following class diagram
+
+```mermaid
+classDiagram
+  Cache <|-- VolumeCache
+  Image <|-- LocalImage
+  LocalImage <|-- CachingImage
+  CachingImage .. VolumeCache
+class Cache {
+    <<interface>>
+}
+
+class Image {
+    <<interface>>
+}
+```
+
+When the Daemon is enabled, a `CachingImage` is created, this image implementation saves the layers tarballs in a `VolumeCache` to reuse them and increase the speed of the process. The idea could be to redesign this `Cache` implementation (VolumeCache) with a new one, maybe a `OCICache` which handles the details to avoid saving the layers tarballs duplicated.
+
+```mermaid
+classDiagram
+  Cache <|-- OCICache
+  Image <|-- LocalImage
+  LocalImage <|-- CachingImage
+  CachingImage .. OCICache
+class Cache {
+    <<interface>>
+
+}
+
+class Image {
+    <<interface>>
+
+}
+```
+### Drawbacks
```

Comment:
Another drawback: the lifecycle still knows about the daemon 😞 . Maybe it's worth circling around and describing the reasons we found the alternative proposals unsatisfying. From what I can recall:

* oci-layout as the interface: lookups (the run image would need to be present on disk)
  * potentially made worse by dynamic run image selection w/ Dockerfiles
* registry-only as the interface: the need for platforms to re-implement daemon loading (pack, maven plugin, others?)
* registry-only as the interface: performance impact of having to write the entire image before loading into the daemon
...probably I missed some things

Reply:
cc @jromero @samj1912

Reply:
After today's WG, I think this RFC is going in the right direction regarding the approaches discussed. 

- Adding the capability to export to OCI format to the Lifecycle opens the door to the other tooling we are considering (prepare, publish, etc)
  - It will also allow other Platforms to prepare their implementation once the daemon support is removed because the mechanism Lifecycle is giving them is the OCI format, so they can actually implement the new *loading into daemon* when this feature is available
  - It will also give us the chance to work in the tooling to help platforms to accomplish the goal